samples/outreach/blogs/serving_blogpost/data_utils.py

```diff
@@ -0,0 +1,204 @@
+  if not os.path.exists(directory):
+    os.makedirs(directory)
+
+  base_url = 'https://www.bgc-jena.mpg.de/wetter/mpi_roof_20{:02d}{}.zip'
+
+  merge_2006_2012 = []
+  for i in range(6, 12):
+    merge_2006_2012.append(base_url.format(i, 'a'))
+    merge_2006_2012.append(base_url.format(i, 'b'))
+
+  get_set_of_data(merge_2006_2012, os.path.join(directory, 'merge_2006_2012.csv'))
+
+  merge_2012_2018 = []
+  for i in range(12, 18):
+    merge_2012_2018.append(base_url.format(i, 'a'))
+    merge_2012_2018.append(base_url.format(i, 'b'))
+  # Special case for present data
+  merge_2012_2018.append('https://www.bgc-jena.mpg.de/wetter/mpi_roof.zip')
+
+  get_set_of_data(merge_2012_2018, os.path.join(directory, 'merge_2012_2018.csv'))
+
+
+def get_set_of_data(urls, output_filename):
+  output_file = open(output_filename, 'wb')
+  for i, url in enumerate(urls):
+    print('Downloading: {}'.format(url))
+    get_single_zip_data(url, output_file, i == 0)
+
+  output_file.close()
+
+
+def get_single_zip_data(url, output_file, first=False):
+  url = urllib.request.urlopen(url)
+  with ZipFile(BytesIO(url.read())) as my_zip_file:
+    contained_file = my_zip_file.namelist()[0]
+    for i, line in enumerate(my_zip_file.open(contained_file).readlines()):
+      if i == 0 and not first or '-9999' in str(line):
+        # Skip in the case of a header (and not the first file) or invalid data
+        continue
+      output_file.write(line)
+
+class Dataset():
+  def __init__(self,
+               data_path='./data/merge_2006_2012.csv',
+               batch_size=128,
+               data_points_day=144,
+               num_days=10,
+               steps=6,
+               mode='train'):
+    self.num_days = num_days
+    self.data_points_day = data_points_day
+    self.data_path = data_path
+    # Look back 10 days
+    self.lookback = data_points_day * num_days
+    # observations once every hour
+    self.steps = steps
+    # We will target one day in the future (24 hours)
+    self.delay = data_points_day
+    self.batch_size = batch_size
+    if mode == 'inference':
+      print('Setting batch size to 1 in inference mode')
+      self.batch_size = 1
+
+  def _preprocess_dataset_element(self, idx, data, lookback, steps, delay):
+    """fn to return the actual data given a specific index"""
+    # We map this function onto individual elements of the dataset
+    sample = data[idx - lookback: idx: steps]
+    target = tf.expand_dims(data[idx + delay][1], -1)
+    return sample, target
+
+  def get_dataset(self,
+                  data,
+                  shuffle=False):
+    """
+    Returns a tf.data.Dataset object containing the data associated with the
+    jena temperature data
+
+    Arguments:
+      data:  raw normalized data
+      shuffle:  boolean true false for shuffling the data or not
+
+    Returns:
+      samples of size (batch, lookback // steps, num features),
+             targets of size (batch,)
+    """
+    num_x = len(data)
```

Comment:
This will work if we change the last line to `ds = ds.map((lambda window: window[:lookback:steps],window[-1]))` 

Do you think this would be more clear? It might be confusing that we grab an additional `delay` worth of data only to grab the last slice? It's certainly less code, so I'll include it for now. Thanks!